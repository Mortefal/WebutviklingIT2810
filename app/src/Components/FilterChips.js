import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Chip from '@material-ui/core/Chip';
import Paper from "@material-ui/core/Paper/Paper";
import Grid from "@material-ui/core/Grid/Grid";
import Typography from "@material-ui/core/Typography/Typography";

const styles = theme => ({
    root:{
        display: 'flex',
        justifyContent: 'flex-start',
        flexWrap: 'wrap',
    },
    chip: {
        margin: theme.spacing.unit,
    }
})
/*const beer = "productType=Sider&productType=Lys ale&productType=Klosterstil&productType=India pale ale&productType=Brown ale&productType=Pale ale&productType=Spesial&productType=Hveteøl&productType=Surøl"
*/
class FilterChips extends React.Component {
    state = {
        filterQuery: [
            {key: 0, label: "Øl og sider"},
            {key: 1, label: "Rødvin"},
            {key: 2, label: "Hvitvin"},
            {key: 3, label: "Annen vin"},
            {key: 4, label: "Musserende vin"},
            {key: 5, label: "Sprit"},
            {key: 6, label: "Alkoholfritt"},
        ],
        filtrationArray: [],
    }
    handleClick = data => () => {
        if (!this.state.filtrationArray.includes(data)) {
            this.setState({filtrationArray: [...this.state.filtrationArray, data]});
        }
    };
    handleDelete = data => () => {
        if(this.state.filtrationArray.includes(data)){
            const chipData = this.state.filtrationArray;
            const chipToDelete = chipData.indexOf(data);
            chipData.splice(chipToDelete, 1);
            this.setState({filtrationArray: [...chipData]});
        }
    };

    render() {
        const {classes} = this.props;

        let alternative = this.state.filterQuery.map(data => {
            let icon = null;
            return (
                <Chip
                    key={data.key}
                    label={data.label}
                    onClick={this.handleClick(data)}
                    className={classes.chip}
                    clickable={true}
                />
            )
        })
        let selected = this.state.filtrationArray.map(data => {
            return (
                <Chip
                    key={data.key}
                    label={data.label}
                    className={classes.chip}
                    clickable={false}
                    color={"secondary"}
                    onDelete={this.handleDelete(data)}
                />)
        });

        return (
            <Paper className={classes.root}>
                <Grid item container>
                    {alternative}
                </Grid>
                <Grid item container>
                    {selected}
                </Grid>
            </Paper>
        )
    }
}
FilterChips.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(FilterChips);

