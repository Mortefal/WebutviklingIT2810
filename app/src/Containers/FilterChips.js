import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Chip from '@material-ui/core/Chip';
import Paper from "@material-ui/core/Paper/Paper";
import Grid from "@material-ui/core/Grid/Grid";

const styles = theme => ({
    root:{
        display: 'flex',
        justifyContent: 'center',
        flexWrap: 'wrap',
        marginBottom: 10,
    },
    chip: {
        margin: theme.spacing.unit,
    }
});

class FilterChips extends React.Component {
    state = {
        filterQuery: [
            {key: 0, label: "Hvitvin"},{key: 1, label:"Musserende vin"},{key: 2, label:"Rødvin"},{key: 3, label:"Rosévin"},
            {key: 4, label:"Fruktvin"},{key: 5, label:"Portvin"},{key: 6, label:"Perlende vin, hvit"},{key: 7, label:"Perlende vin, rød"},
            {key: 8, label:"Perlende vin, rosé"},{key: 9, label:"Sterkvin, annen"},
            {key: 10, label:"Sherry"},{key: 11, label:"Vermut"},{key: 12, label:"Madeira"},
        ],
        filtrationArray: [],
    };

    handleClick = data => () => {
        if (!this.state.filtrationArray.includes(data)) {
            this.setState({filtrationArray: [data]});
        }
        this.props.callback(data.label)
    };

    handleDelete = data => () => {
        if(this.state.filtrationArray.includes(data)){
            const chipData = this.state.filtrationArray;
            const chipToDelete = chipData.indexOf(data);
            chipData.splice(chipToDelete, 1);
            this.setState({filtrationArray: [...chipData]});
        }
        this.props.callback(null)
    };

    render() {
        const {classes} = this.props;

        let alternative = this.state.filterQuery.map(data => {
            return (
                <Chip
                    key={data.key}
                    label={data.label}
                    onClick={this.handleClick(data)}
                    className={classes.chip}
                    clickable={true}
                />
            )
        });

        let selected = this.state.filtrationArray.map(data => {
            return (
                <Chip
                    key={data.key}
                    label={data.label}
                    className={classes.chip}
                    clickable={false}
                    color={"secondary"}
                    onDelete={this.handleDelete(data)}
                />)
        });

        return (
            <Paper className={classes.root}>
                <Grid item container>
                    {alternative}
                </Grid>
                <Grid item container>
                    {selected}
                </Grid>
            </Paper>
        )
    }
}

FilterChips.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(FilterChips);
