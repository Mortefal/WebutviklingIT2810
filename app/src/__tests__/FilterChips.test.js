import React from 'react';
import FilterChips from '../Containers/FilterChips';
import renderer from 'react-test-renderer'


describe('Snapshot test of FilterChips', ()=> {
    it('renders correctly', () => {
        const tree = renderer
            .create(<FilterChips/>)
            .toJSON();
        expect(tree).toMatchSnapshot()
    })
});
