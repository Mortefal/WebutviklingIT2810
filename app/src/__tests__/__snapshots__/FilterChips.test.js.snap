// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of DetailsPage renders correctly 1`] = `
<div
  className="MuiPaper-root-3 MuiPaper-elevation2-7 MuiPaper-rounded-4 FilterChips-root-1"
>
  <div
    className="MuiGrid-container-30 MuiGrid-item-31"
  >
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Øl og sider
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Rødvin
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Hvitvin
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Annen vin
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Musserende vin
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Sprit
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Alkoholfritt
      </span>
    </div>
  </div>
  <div
    className="MuiGrid-container-30 MuiGrid-item-31"
  />
</div>
`;

exports[`Snapshot test of FilterChips renders correctly 1`] = `
<div
  className="MuiPaper-root-3 MuiPaper-elevation2-7 MuiPaper-rounded-4 FilterChips-root-1"
>
  <div
    className="MuiGrid-container-30 MuiGrid-item-31"
  >
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Øl og sider
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Rødvin
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Hvitvin
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Annen vin
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Musserende vin
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Sprit
      </span>
    </div>
    <div
      className="MuiChip-root-127 MuiChip-clickable-130 FilterChips-chip-2"
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiChip-label-146"
      >
        Alkoholfritt
      </span>
    </div>
  </div>
  <div
    className="MuiGrid-container-30 MuiGrid-item-31"
  />
</div>
`;
